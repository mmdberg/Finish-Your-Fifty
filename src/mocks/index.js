export const mockRace = {
  venue: 'Atlas Park',
  city: 'San Diego',
  event: 'Spooky 5k',
  date: '04-05-2018'
};

export const mockApiResult = {
  place: {
    placeName: 'Atlas Park',
    cityName: 'San Diego'
  },
  activityStartDate: '2018-04-05T08:00:00',
  organization: {
    organizationName: 'Spooky 5k'
  }
};

export const mockUser = {
  name: 'Pizza',
  email: 'pizza@pizza',
  password: 'p'
};