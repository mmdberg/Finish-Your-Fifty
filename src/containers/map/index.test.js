import { shallow } from 'enzyme';
import { StateMap } from './index';
import React from 'react';
import {mockCompletedRace} from '../../mocks';

describe('Map', () => {

  it('should match snapshot', () => {
    let wrapper = shallow(<StateMap races={[mockCompletedRace]}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should correctly map races to props', () => {

    });
  });
});