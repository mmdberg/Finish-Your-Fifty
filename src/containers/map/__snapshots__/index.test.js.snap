// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StateMap
    races={
        Array [
            Object {
              "city": "Nashville",
              "completed": "Completed",
              "date": "04-05-2018",
              "distance": "Marathon",
              "error": "",
              "raceName": "Hocus Marathon",
              "state": "TN",
              "time": "4:00:01",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="map-key"
>
          <p>
                    You have run 
                    1
                     races in 
                    1
                     states!
          </p>
          <p>
                    Click on states to see race highlights.
          </p>
          <p>
                    Click below filter by distance:
          </p>
          <Dropdown
                    baseClassName="Dropdown"
                    name="distance"
                    onChange={[Function]}
                    options={
                              Array [
                                        "All Races",
                                        "Marathon",
                                        "Half Marathon",
                                        "10 Miler",
                                        "10K",
                                        "5K",
                                        "Other",
                                      ]
                    }
                    placeholder="All Races"
                    value=""
          />
</div>,
        <div
          className="state-map"
>
          <div
                    className="raceInfo"
          >
                    <p
                              className="raceText"
                              style={
                                        Object {
                                                  "display": "none",
                                                  "left": 0,
                                                  "top": 0,
                                                }
                              }
                    />
          </div>
          <USAMap
                    customize={
                              Object {
                                        "TN": Object {
                                          "fill": "rgba(144, 63, 63, 0.75)",
                                        },
                                      }
                    }
                    defaultFill="#D3D3D3"
                    height={593}
                    onClick={[Function]}
                    title="Blank US states map"
                    width={959}
          />
</div>,
      ],
      "className": "map",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              You have run 
              1
               races in 
              1
               states!
</p>,
            <p>
              Click on states to see race highlights.
</p>,
            <p>
              Click below filter by distance:
</p>,
            <Dropdown
              baseClassName="Dropdown"
              name="distance"
              onChange={[Function]}
              options={
                            Array [
                                          "All Races",
                                          "Marathon",
                                          "Half Marathon",
                                          "10 Miler",
                                          "10K",
                                          "5K",
                                          "Other",
                                        ]
              }
              placeholder="All Races"
              value=""
/>,
          ],
          "className": "map-key",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "You have run ",
                1,
                " races in ",
                1,
                " states!",
              ],
            },
            "ref": null,
            "rendered": Array [
              "You have run ",
              1,
              " races in ",
              1,
              " states!",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click on states to see race highlights.",
            },
            "ref": null,
            "rendered": "Click on states to see race highlights.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click below filter by distance:",
            },
            "ref": null,
            "rendered": "Click below filter by distance:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "baseClassName": "Dropdown",
              "name": "distance",
              "onChange": [Function],
              "options": Array [
                "All Races",
                "Marathon",
                "Half Marathon",
                "10 Miler",
                "10K",
                "5K",
                "Other",
              ],
              "placeholder": "All Races",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="raceInfo"
>
              <p
                            className="raceText"
                            style={
                                          Object {
                                                        "display": "none",
                                                        "left": 0,
                                                        "top": 0,
                                                      }
                            }
              />
</div>,
            <USAMap
              customize={
                            Object {
                                          "TN": Object {
                                            "fill": "rgba(144, 63, 63, 0.75)",
                                          },
                                        }
              }
              defaultFill="#D3D3D3"
              height={593}
              onClick={[Function]}
              title="Blank US states map"
              width={959}
/>,
          ],
          "className": "state-map",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="raceText"
                style={
                                Object {
                                                "display": "none",
                                                "left": 0,
                                                "top": 0,
                                              }
                }
/>,
              "className": "raceInfo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "raceText",
                "style": Object {
                  "display": "none",
                  "left": 0,
                  "top": 0,
                },
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "customize": Object {
                "TN": Object {
                  "fill": "rgba(144, 63, 63, 0.75)",
                },
              },
              "defaultFill": "#D3D3D3",
              "height": 593,
              "onClick": [Function],
              "title": "Blank US states map",
              "width": 959,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="map-key"
>
            <p>
                        You have run 
                        1
                         races in 
                        1
                         states!
            </p>
            <p>
                        Click on states to see race highlights.
            </p>
            <p>
                        Click below filter by distance:
            </p>
            <Dropdown
                        baseClassName="Dropdown"
                        name="distance"
                        onChange={[Function]}
                        options={
                                    Array [
                                                "All Races",
                                                "Marathon",
                                                "Half Marathon",
                                                "10 Miler",
                                                "10K",
                                                "5K",
                                                "Other",
                                              ]
                        }
                        placeholder="All Races"
                        value=""
            />
</div>,
          <div
            className="state-map"
>
            <div
                        className="raceInfo"
            >
                        <p
                                    className="raceText"
                                    style={
                                                Object {
                                                            "display": "none",
                                                            "left": 0,
                                                            "top": 0,
                                                          }
                                    }
                        />
            </div>
            <USAMap
                        customize={
                                    Object {
                                                "TN": Object {
                                                  "fill": "rgba(144, 63, 63, 0.75)",
                                                },
                                              }
                        }
                        defaultFill="#D3D3D3"
                        height={593}
                        onClick={[Function]}
                        title="Blank US states map"
                        width={959}
            />
</div>,
        ],
        "className": "map",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                You have run 
                1
                 races in 
                1
                 states!
</p>,
              <p>
                Click on states to see race highlights.
</p>,
              <p>
                Click below filter by distance:
</p>,
              <Dropdown
                baseClassName="Dropdown"
                name="distance"
                onChange={[Function]}
                options={
                                Array [
                                                "All Races",
                                                "Marathon",
                                                "Half Marathon",
                                                "10 Miler",
                                                "10K",
                                                "5K",
                                                "Other",
                                              ]
                }
                placeholder="All Races"
                value=""
/>,
            ],
            "className": "map-key",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "You have run ",
                  1,
                  " races in ",
                  1,
                  " states!",
                ],
              },
              "ref": null,
              "rendered": Array [
                "You have run ",
                1,
                " races in ",
                1,
                " states!",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Click on states to see race highlights.",
              },
              "ref": null,
              "rendered": "Click on states to see race highlights.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Click below filter by distance:",
              },
              "ref": null,
              "rendered": "Click below filter by distance:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "baseClassName": "Dropdown",
                "name": "distance",
                "onChange": [Function],
                "options": Array [
                  "All Races",
                  "Marathon",
                  "Half Marathon",
                  "10 Miler",
                  "10K",
                  "5K",
                  "Other",
                ],
                "placeholder": "All Races",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="raceInfo"
>
                <p
                                className="raceText"
                                style={
                                                Object {
                                                                "display": "none",
                                                                "left": 0,
                                                                "top": 0,
                                                              }
                                }
                />
</div>,
              <USAMap
                customize={
                                Object {
                                                "TN": Object {
                                                  "fill": "rgba(144, 63, 63, 0.75)",
                                                },
                                              }
                }
                defaultFill="#D3D3D3"
                height={593}
                onClick={[Function]}
                title="Blank US states map"
                width={959}
/>,
            ],
            "className": "state-map",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="raceText"
                  style={
                                    Object {
                                                      "display": "none",
                                                      "left": 0,
                                                      "top": 0,
                                                    }
                  }
/>,
                "className": "raceInfo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "raceText",
                  "style": Object {
                    "display": "none",
                    "left": 0,
                    "top": 0,
                  },
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "customize": Object {
                  "TN": Object {
                    "fill": "rgba(144, 63, 63, 0.75)",
                  },
                },
                "defaultFill": "#D3D3D3",
                "height": 593,
                "onClick": [Function],
                "title": "Blank US states map",
                "width": 959,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
